WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -9,9 +10,17 @@ def bad_api_parameter(e)
 
   DEFAULT_API_RESULT_LIMIT = 20
 
+  def current_resource_owner
+    if doorkeeper_token
+      User.find_by_id(doorkeeper_token.resource_owner_id).tap do |user|
+        user.doorkeeper_token = doorkeeper_token
```

Comment:
Ah, good point! Fixed in 1a0ce982b28587d79fc411ab76f5443282873df9. There are a lot of ways of skinning this cat, so I'm open to alternative implementations if someone feels there's a better way.